src/iter.rs

```diff
@@ -226,17 +239,29 @@ impl ExactSizeIterator for HexDigitsIter<'_> {}
 
 impl core::iter::FusedIterator for HexDigitsIter<'_> {}
 
-/// `hi` and `lo` are bytes representing hex characters.
+/// `hi` and `lo` are bytes that encode hex digits as ASCII.
 ///
-/// Returns the valid byte or the invalid input byte and a bool indicating error for `hi` or `lo`.
-fn hex_chars_to_byte(hi: u8, lo: u8) -> Result<u8, (u8, bool)> {
-    let hih = (hi as char).to_digit(16).ok_or((hi, true))?;
-    let loh = (lo as char).to_digit(16).ok_or((lo, false))?;
+/// Returns the valid byte or a bool indicating error for `hi` or `lo`.
+fn hex_digits_to_byte(hi: u8, lo: u8) -> Result<u8, bool> {
```

Comment:
In ce28e9ba01cda818439b36108e072435715552fc:

I find the word "digit" when used to apply to a *number*, when we are working with a character encoding that contains actual digits, is extremely confusing.

Reply:
Amusingly I don't even know what `hi` and `lo` are from just looking at this diff and I worked on this code yesterday for hours. I thought 'symbol' instead of 'digit' but perhaps 'hex_ascii_value_to_byte` but the perhaps `to_byte` is actually better described by 'decode' because the two hex symbols encode a byte. 

Perhaps: `decode_hex_ascii_value_pair`

Reply:
Yep, `ascii_value_pair` sounds good to me.

And yeah, `hi` and `lo` are not great... they are obviously the two nybbles of the byte, but are they ASCII values or numbers? Not easy to tell when we are using `u8` for both.